From 39f54746462b274de4f0d186cd3db88d75edc95d Mon Sep 17 00:00:00 2001
From: Simon Zolin <4729655+stsaz@users.noreply.github.com>
Date: Sat, 22 Aug 2026 00:00:00 +0300
Subject: [PATCH] + x86 filter

---
 src/liblzma/simple/x86.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/src/liblzma/simple/x86.c b/src/liblzma/simple/x86.c
index dffa786..919127f 100644
--- a/src/liblzma/simple/x86.c
+++ b/src/liblzma/simple/x86.c
@@ -19,6 +19,7 @@
 typedef struct {
 	uint32_t prev_mask;
 	uint32_t prev_pos;
+	uint32_t now_pos;
 } lzma_simple_x86;
 
 
@@ -179,3 +180,21 @@ lzma_bcj_x86_decode(uint32_t start_offset, uint8_t *buf, size_t size)
 	return x86_code(&simple, start_offset, false, buf, size);
 }
 #endif
+
+
+#include <lzma-ff.h>
+
+static size_t lzma_x86_decode(void *ctx, char *buf, size_t size)
+{
+	lzma_simple_x86 *c = ctx;
+	size_t n = x86_code(c, c->now_pos, 0, (void*)buf, size);
+	c->now_pos += n;
+	return n;
+}
+
+const lzma_coder_ctx lzma_x86_ctx = {
+	.method = LZMA_FILTER_X86,
+	.ctxsize = sizeof(lzma_simple_x86),
+	.max_unprocessed = 5,
+	.simple_decoder = &lzma_x86_decode,
+};
-- 
2.53.0

